modules/tfm/tfm/boards/common/assert.c

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <pm_config.h>
+
+
+
+/* The size of of the tfm partition assumes the hardcoded size of of 0x200 for the mcuboot_pad */
+/* partition. So this is checked during the build in case it got modified afterwards */
+#if (defined(PM_MCUBOOT_PAD_SIZE) && PM_MCUBOOT_PAD_SIZE!=0x200)
```

Comment:
We will never be able to compile something with MCUBOOT_PAD_SIZE != 0x200 now. 

I think we should just add asserts that the non-secure partitions has ended up being aligned, and let the application writer figure out how this can be achieved with the various sizes preceding.

Something like this:
```
#include "pm_config.h"
#include "autoconf.h"

#define IS_ALIGNED_POW2(value, align) (((value) & ((align) - 1)) == 0)

#if !(IS_ALIGNED_POW2(PM_TFM_NONSECURE_ADDRESS, CONFIG_NRF_SPU_FLASH_REGION_SIZE))
#error "TF-M non-secure address start is not aligned on SPU region size"
#endif

#if !(IS_ALIGNED_POW2(PM_TFM_NONSECURE_SIZE, CONFIG_NRF_SPU_FLASH_REGION_SIZE))
#error "TF-M non-secure address size is not aligned on SPU region size"
#endif
```

Should be repeatet for the other regions that we configure.
For flash this is:
nonsecure storage
secondary partition
And for RAM:
SRAM_NONSECURE

Reply:
I'm noticing that the assert on size will be triggered for some of these configurations, but it should be acceptable because of  secondary partition following it.
Maybe we could have it as an extra check, or drop the assert on sizes.

Reply:
Ok good point, 

We don't configure the secondary_partition, just the app/tfm_nonsecure